cloneContribFiles.sh

```diff
@@ -1,7 +1,9 @@
 if [ "$2" = "yes" ]
     then
         rm -rf src/main/groovy/edu/oregonstate/mist/contrib/
-        git clone --depth=1 -q $1 src/main/groovy/edu/oregonstate/mist/contrib
-        rm -rf src/main/groovy/edu/oregonstate/mist/contrib/.git/
-        rm src/main/groovy/edu/oregonstate/mist/contrib/.gitignore
+        git clone -q $1 src/main/groovy/edu/oregonstate/mist/contrib
+        cd src/main/groovy/edu/oregonstate/mist/contrib
+        git reset --hard $3
+        rm -rf .git/
+        rm .gitignore
```

Comment:
It's good practice to put quotes around all shell variables to prevent unexpected word splitting.

    rm -rf src/main/groovy/edu/oregonstate/mist/contrib/ &&
    git clone --quiet "$1" src/main/groovy/edu/oregonstate/mist/contrib &&
    cd src/main/groovy/edu/oregonstate/mist/contrib &&
    git reset --hard "$3" &&
    rm -rf .git/ .gitignore